prusti-viper/src/encoder/mir/pure/pure_functions/interpreter.rs

```diff
@@ -830,7 +828,8 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
                                     let (encoded_rhs, is_value) = self.encode_operand(operand).with_span(span)?;
                                     if is_value {
```

Comment:
It is now true that in this file that all pure function arguments are passed as values (i.e. primitive values or snapshot values)?

Reply:
I think yes, so the second element of the `encode_operand` tuple should be removable.

Reply:
Then, I think that the `state.substitute_value(&encoded_lhs, snapshot);` at (new) line 846 should be enough to encode this case. That is, the `state.substitute_value(&field_place, encoded_rhs.clone());` should be redundant because `encoded_lhs` is always going to be a proper prefix of `field_place`. The same should hold for many other cases (`AggregateKind::Adt` etc).